src/Provider/AbstractProvider.php

```diff
@@ -766,7 +766,15 @@ protected function fetchResourceOwnerDetails(AccessToken $token)
 
         $request = $this->getAuthenticatedRequest(self::METHOD_GET, $url, $token);
 
-        return $this->getParsedResponse($request);
+        $response = $this->getParsedResponse($request);
+
+        if (false === is_array($response)) {
+            throw new UnexpectedValueException(
+                'Invalid response received from Authorization Server. Expected JSON.'
```

Comment:
I agree to that.  In fact, we do have the URL that was being accessed in that block.  Instead of Authorization server we could use this value.